crates/ruff/src/rules/flake8_pyi/rules/import_from_future.rs

```diff
@@ -0,0 +1,32 @@
+use rustpython_parser::ast::StmtImportFrom;
+
+use ruff_diagnostics::{Diagnostic, Violation};
+use ruff_macros::{derive_message_formats, violation};
+
+use crate::checkers::ast::Checker;
+
+#[violation]
+pub struct ImportFromFuture;
+
+impl Violation for ImportFromFuture {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        format!("`from __future__` import annotations has no effect in stub files, since type checkers automatically treat stubs as having those semantics")
```

Comment:
```suggestion
        format!("`from __future__ import annotations` has no effect in stub files, since type checkers automatically treat stubs as having those semantics")
```